sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -2358,7 +2367,21 @@ public void delete(String caseUuid) throws ExternalSurveillanceToolException {
 		deleteCase(caze);
 	}
 
+	@Override
+	public void deleteWithContacts(String caseUuid) {
+
+		if (!userService.hasRight(UserRight.CASE_DELETE)) {
+			throw new UnsupportedOperationException("User " + userService.getCurrentUser().getUuid() + " is not allowed to delete cases.");
+		}
+
+		Case caze = service.getByUuid(caseUuid);
+		deleteCase(caze);
+
+		Optional.of(caze.getContacts()).ifPresent(cl -> cl.forEach(c -> contactService.delete(c)));
+	}
+
 	private void deleteCase(Case caze) throws ExternalSurveillanceToolException {
+
 		externalJournalService.handleExternalJournalPersonUpdateAsync(caze.getPerson().toReference());
 		if (externalSurveillanceToolGatewayFacade.isFeatureEnabled() && caze.getExternalID() != null && !caze.getExternalID().isEmpty()) {
 			List<CaseDataDto> casesWithSameExternalId = getByExternalId(caze.getExternalID());
```

Comment:
@MateStrysewske The deletion in external surveillance tool also has to be executed on permanent deletion. I forgot to put this into the issue.